strapi/strapi
# [POC] Fix the failing CI tests on Node 20 / sqlite3

### What does it do?

Update: I'm actually not sure this is the main problem and it may be a side effect of something else.

Demos the problem with the failing CI test for Node 20 / sqlite3. Essentially, Node 20 with the sqlite3 package (not better-sqlite3) is inserting Date fields as null unless converted to string first.

Note: this isn't a suggested solution, just a POC to discuss the right solution

And I'm not sure why this wasn't failing when node 20 was merged, because it was running there. I'm wondering if something changed regarding dates somewhere recently.

### How to test it?

I've commented out the lines that would prefer better-sqlite3, so just use Node 20 to run a test that uses dates, like `yarn test:api migration-draft-publish.test.api.js` and see that it now passes with this fix and fails without it.
